airbyte-integrations/connectors/source-outplay/source_outplay/stream.py

```diff
@@ -0,0 +1,377 @@
+            params["fieldcondition"] = self.cursor_field_condition
+
+        return params
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return {self.cursor_field: self._cursor_value}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = value[self.cursor_field]
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        records = super().read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state)
+        for record in records:
+            blank_val = 0 if self.is_integer_state else ""
+            self._cursor_value = (
+                max(record.get(self.cursor_field, blank_val), self._cursor_value if not self.is_integer_state else int(self._cursor_value))
+                if self._cursor_value is not None
+                else record.get(self.cursor_field, blank_val)
+            )
+            yield record
+
+    def filter_records_newer_than_state(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        records_slice: Mapping[str, Any] = None,
+    ) -> Iterable:
+        if self._cursor_value is not None and records_slice is not None:
+            for record in records_slice:
+                if record[self.cursor_field] >= self._cursor_value:
+                    yield record
+        elif records_slice is not None:
+            yield from records_slice
+
+
+class Prospects(OutplayIncrementalReplicationStream):
+    """
+    API documentation: https://documenter.getpostman.com/view/16947449/TzsikPV1#0bb4bb2d-6b8b-45d2-932e-588131c69e24
+    """
+
+    use_cache = True
+    object_name = "prospect/search"
+
+
+# OutplayReportDateIncrementalReplicationStream
+class OutplayReportDateIncrementalReplicationStream(OutplayStream, IncrementalMixin):
+    http_method = "GET"
+    cursor_field = "ReportDate"
+    date_filter_template = "%Y-%m-%d"
+    additional_filters: Optional[Mapping[str, Any]] = None
+    state_checkpoint_interval = 1000
+    _cursor_value: Optional[str] = None
+    _last_request_cursor_value: Optional[date] = None
+
+    def _is_page_token_available(self, next_page_token: Optional[Mapping[str, Any]]):
+        return next_page_token is not None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        results = response.json()
+        report_date = results[0].get("report_date", None) if isinstance(results, list) and len(results) > 0 else None
+
+        if report_date is None:
+            report_date = self._last_request_cursor_value or pendulum.parse(self.config.get("start_date", None), strict=False).date()
+
+        next_date = report_date + timedelta(days=1)
+        max_api_requests = self.config["max_api_requests"] or 10000
```

Comment:
Why is max 10000, but in api.py its 100000?

Maybe make a constant somewhere if they should match?